mopidy/core/library.py

```diff
@@ -313,8 +332,14 @@ def search(self, query=None, uris=None, exact=False, **kwargs):
 
         results = []
         for backend, future in futures.items():
-            try:
-                results.append(future.get())
+            try:  # TODO: fix all these cases so we can use common helper
```

Comment:
Can't you except TypeError and LookupError inside the `_backend_error_handling` context, and let everything else (ValidationError and Exception) bubble up to the context manager?